libraries/lib-wave-track/PasteOverPreservingClips.cpp

```diff
@@ -76,21 +76,23 @@ void PasteOverPreservingClips(const ClipData &data,
       //remove the old audio and get the NEW
       auto [start, end] = clipStartEndTimes[i];
       oldTrack.Clear(start, end);
-   
+
       auto toClipOutput = newContents.Copy(start - startT, end - startT);
       oldTrack.Paste(start, *toClipOutput);
 
       //Restore original clip's name
       auto newClip = oldTrack.GetClipAtTime(start + 0.5 / oldTrack.GetRate());
       newClip->SetName(clipNames[i]);
 
-      //if the clip was only partially selected, the Paste will have created a
-      // split line.  Join is needed to take care of this
-      //This is not true when the selection is fully contained within one clip
-      // (second half of conditional)
+      // The Paste will have created a split line. Join is needed to take care
+      // of this.
       auto [realStart, realEnd] = clipRealStartEndTimes[i];
-      if ((realStart  != start || realEnd != end) &&
-         !(realStart <= startT && realEnd >= startT + lenT))
-         oldTrack.Join(realStart, realEnd);
+      oldTrack.Join(
```

Comment:
A conditional call to `Join` is now unconditional.  Does this change behavior?

In the case that the selected portion of track for EQ contains split lines, not at the extremes of the selection, then this is intended to put the lines back where they were.  Does this conserve that behavior?

Reply:
I think that `clipRealStartEndTimes` contain the original clip boundaries, and `Join` isn't called across any of these.

But I'm trying to figure out if this function is needed at all. What it does:
Takes `newContents`, a one-clip track with the output of the effect, replaces the content of `oldTrack` in `region = [start, start + len)` with it, preserving existing clip boundaries strictly within `region`, but merging at `start` if it splits a clip underneath, and likewise for `start + len`.
This is the intended behaviour for other effects - if merging doesn't mean stretch-rendering, else there is no merging. Can you recall why this is a special case ?

Also, it crashes on `master` for [this project](https://github.com/audacity/audacity/files/12657643/test.zip): not modifying the selected region, apply some eq. `GetClipAtTime` above returns an unchecked nullptr. I haven't got to the bottom of this but if this function isn't needed I may not have to. #5138 doesn't repair it but this branch does somehow.